File: Other/N2F/Src/Next2Friends.Social/src/j2me/src/service/DashboardService.java
package service;
import java.io.IOException;

public class DashboardService extends BaseService
{
    public DashboardNewFriend[] GetNewFriends(String nickname, String password) throws IOException
    {
        prepare(18);

        pos.writeString(nickname);
        pos.writeString(password);

        try
        {
            commit();
        }
        catch(IOException ex)
        {
            throw ex;
        }
  
        DashboardNewFriend[] res = pis.readDashboardnewfriendArray();
        return res;
    }

    public DashboardPhoto[] GetPhotos(String nickname, String password) throws IOException
    {
        prepare(19);

        pos.writeString(nickname);
        pos.writeString(password);

        try
        {
            commit();
        }
        catch(IOException ex)
        {
            throw ex;
        }
  
        DashboardPhoto[] res = pis.readDashboardphotoArray();
        return res;
    }

    public DashboardVideo[] GetVideos(String nickname, String password) throws IOException
    {
        prepare(20);

        pos.writeString(nickname);
        pos.writeString(password);

        try
        {
            commit();
        }
        catch(IOException ex)
        {
            throw ex;
        }
  
        DashboardVideo[] res = pis.readDashboardvideoArray();
        return res;
    }

    public DashboardWallComment[] GetWallComments(String nickname, String password) throws IOException
    {
        prepare(21);

        pos.writeString(nickname);
        pos.writeString(password);

        try
        {
            commit();
        }
        catch(IOException ex)
        {
            throw ex;
        }
  
        DashboardWallComment[] res = pis.readDashboardwallcommentArray();
        return res;
    }

}

File: Other/N2F/Src/Next2Friends.Social/src/independent/source/services/DashboardService.cpp
#include "DashboardService.h"
#include "Server.h"
#include "PacketStream.h"
#include "AskQuestionConfirm.h"
#include "AskQuestion.h"
#include "AskComment.h"
#include "AskResponse.h"
#include "StringArray.h"
#include "ArrayOfDashboardVideo.h"
#include "DashboardVideo.h"
#include "DashboadMedia.h"
#include "ArrayOfDashboardPhoto.h"
#include "DashboardPhoto.h"
#include "ArrayOfDashboardWallComment.h"
#include "DashboardWallComment.h"
#include "ArrayOfDashboardNewFriend.h"
#include "DashboardNewFriend.h"


DashboardService::DashboardService( Server *server )
: BaseService(server)
{
}

DashboardService::~DashboardService()
{

}

bool DashboardService::GetNewFriends(IAnswer *answerCB, char16* nickname, char16* password)
{
    if (!StartRequest(answerCB, DashboardService::EPS_GETNEWFRIENDS))
    {
        return false;
    }

    PacketStream *str = pServer->GetStream();
    bool isOk = true;

    isOk = isOk && str->WriteInt32(18);//request id

  
    isOk = isOk &&  str->WriteWString(nickname);
    isOk = isOk &&  str->WriteWString(password);

    return isOk && EndRequest();

}

ArrayOfDashboardNewFriend *DashboardService::OnGetNewFriends()
{
  
  ArrayOfDashboardNewFriend *res = new ArrayOfDashboardNewFriend();
    res->ReadFromStream(answerStream);

    return res;
}


bool DashboardService::GetPhotos(IAnswer *answerCB, char16* nickname, char16* password)
{
    if (!StartRequest(answerCB, DashboardService::EPS_GETPHOTOS))
    {
        return false;
    }

    PacketStream *str = pServer->GetStream();
    bool isOk = true;

    isOk = isOk && str->WriteInt32(19);//request id

  
    isOk = isOk &&  str->WriteWString(nickname);
    isOk = isOk &&  str->WriteWString(password);

    return isOk && EndRequest();

}

ArrayOfDashboardPhoto *DashboardService::OnGetPhotos()
{
  
  ArrayOfDashboardPhoto *res = new ArrayOfDashboardPhoto();
    res->ReadFromStream(answerStream);

    return res;
}


bool DashboardService::GetVideos(IAnswer *answerCB, char16* nickname, char16* password)
{
    if (!StartRequest(answerCB, DashboardService::EPS_GETVIDEOS))
    {
        return false;
    }

    PacketStream *str = pServer->GetStream();
    bool isOk = true;

    isOk = isOk && str->WriteInt32(20);//request id

  
    isOk = isOk &&  str->WriteWString(nickname);
    isOk = isOk &&  str->WriteWString(password);

    return isOk && EndRequest();

}

ArrayOfDashboardVideo *DashboardService::OnGetVideos()
{
  
  ArrayOfDashboardVideo *res = new ArrayOfDashboardVideo();
    res->ReadFromStream(answerStream);

    return res;
}


bool DashboardService::GetWallComments(IAnswer *answerCB, char16* nickname, char16* password)
{
    if (!StartRequest(answerCB, DashboardService::EPS_GETWALLCOMMENTS))
    {
        return false;
    }

    PacketStream *str = pServer->GetStream();
    bool isOk = true;

    isOk = isOk && str->WriteInt32(21);//request id

  
    isOk = isOk &&  str->WriteWString(nickname);
    isOk = isOk &&  str->WriteWString(password);

    return isOk && EndRequest();

}

ArrayOfDashboardWallComment *DashboardService::OnGetWallComments()
{
  
  ArrayOfDashboardWallComment *res = new ArrayOfDashboardWallComment();
    res->ReadFromStream(answerStream);

    return res;
}


